src/kademlia/node.cpp

```diff
@@ -220,7 +220,10 @@ void node::bootstrap(std::vector<udp::endpoint> const& nodes
 
 	for (auto const& n : nodes)
 	{
-		if (n.protocol() != protocol()) continue;
+#if !TORRENT_USE_IPV6
+		if (n.protocol() == udp::v6()) continue;
```

Comment:
The problem is that the state (`dht_state`) have a vector of `udp::endpoint` (two to be precise), there is no really a way to make sure they are all IPv4. This is a safety measure.